handlers/kinetic_request_ce_attachment_copy_remote_server_v1/handler/init.rb

```diff
@@ -0,0 +1,210 @@
+    destination_field_name      = @parameters["destination_field_name"]
+    #Other Variables
+    error_handling  = @parameters["error_handling"]    
+    imported_files = []
+    imported_file_details = []
+
+     # Submission API Route including Values
+      submission_api_route = source_server +
+                             "/app/api/v1/submissions/" +
+                             URI.escape(source_submission_id) +
+                             "/?include=values"
+        puts "Getting from submission: #{submission_api_route}"
+      # Retrieve the Submission Values
+      submission_result = RestClient::Resource.new(
+        submission_api_route,
+        user: source_user,
+        password: source_pass
+      ).get
+
+      puts "Got from submission: #{submission_api_route}"
+      # If the submission exists
+      unless submission_result.nil?
+        submission = JSON.parse(submission_result)["submission"]
+        field_value = submission["values"][source_field_name]
+        # If the attachment field value exists
+        unless field_value.nil?
+          files = []
+          # Attachment field values are stored as arrays, one map for each file attachment
+          field_value.each_index do |index|
+            file_info = field_value[index]
+            tmp_file_name = File.join(@tmp_dir, file_info['name'])
+            # The attachment file name is stored in the 'name' property
+            # API route to get the generated attachment download link from Kinetic Request CE.
+            # "/{spaceSlug}/app/api/v1/submissions/{submissionId}/files/{fieldName}/{fileIndex}/{fileName}/url"
+
+            attachment_download_api_route = source_server +
+              '/app/api/v1' +
+              '/submissions/' + URI.escape(source_submission_id) +
+              '/files/' + URI.escape(source_field_name) +
+              '/' + index.to_s +
+              '/' + URI.escape(file_info['name']) +
+              '/url'
+            puts "Getting attachment from submission: #{file_info['name']} from field #{source_field_name}" if @enable_debug_logging
+            # Retrieve the URL to download the attachment from Kinetic Request CE.
+            # This URL will only be valid for a short amount of time before it expires
+            # (usually about 5 seconds).
+            attachment_download_result = RestClient::Resource.new(
+              attachment_download_api_route,
+              user: source_user,
+              password: source_pass
+            ).get
+
+            unless attachment_download_result.nil?
+                # get the filehub url to download the file
+                fileUrl = JSON.parse(attachment_download_result)['url']
+                puts "Downloading file: #{file_info['name']} from #{fileUrl}" if @enable_debug_logging
+
+                puts "Downloading file to memory" if @enable_debug_logging
+                file_content = RestClient.get(fileUrl).to_java_bytes
```

Comment:
I  can see this becoming a problem. Should stream the response instead of reading the entire file into memory. See https://github.com/rest-client/rest-client/tree/1.6-legacy#label-Lower-level+access